tap_hubspot/hubspot.py

```diff
@@ -223,6 +225,174 @@ def streams(self, start_date: datetime, end_date: datetime):
         else:
             raise NotImplementedError(f"unknown stream_id: {self.tap_stream_id}")
 
+    def get_contacts(
+        self, start_date: datetime, end_date: datetime
+    ) -> Iterable[Tuple[Dict, datetime]]:
+        filter_key = "lastmodifieddate"
+        obj_type = "contacts"
+        gen = self.search(
+            obj_type,
+            filter_key,
+            start_date,
+            end_date,
+            MANDATORY_PROPERTIES[f"{obj_type}"],
+        )
+
+        while True:
+            deals = {}
+            i = 0
+
+            for rec, ts in gen:
+                rec_id = rec["id"]
+                # ensure that there is a default entry for all contacts/companies
+                rec["associations"] = {
+                    # fit the legacy format to be backwards compatible:
+                    # "results": [{ "id": "<id>" }]
+                    "deals": {"results": []},
```

Comment:
Ahh, just spoke with Jing, and since we are already getting the deals -> contacts relation when retrieving deals, there is no need to get the inverse relation here as well (and the production datamodeling code is already relying on the deals relation), so, we only want companies here.